app/src/sharedTest/java/org/oppia/app/utility/OrientationChangeAction.kt

```diff
@@ -0,0 +1,51 @@
+package org.oppia.app.utility
+
+import android.view.View
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.matcher.ViewMatchers.isRoot
+import org.hamcrest.Matcher
+import android.content.pm.ActivityInfo
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
+import android.view.ViewGroup
+
+// https://gist.github.com/nbarraille/03e8910dc1d415ed9740
+/** This class helps in managing orientation changes in test-cases which are visible on device too. */
+class OrientationChangeAction(private val orientation: Int) : ViewAction {
+
+  companion object {
+    fun orientationLandscape(): ViewAction = OrientationChangeAction(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE)
+    fun orientationPortrait(): ViewAction = OrientationChangeAction(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT)
+  }
+
+  override fun getDescription(): String = "change orientation to $orientation"
+
+  override fun getConstraints(): Matcher<View> = isRoot()
+
+  override fun perform(uiController: UiController, view: View) {
+    uiController.loopMainThreadUntilIdle()
+    var activity = getActivity(view.context)
+    if (activity == null && view is ViewGroup) {
+      val c = view.childCount
+      var i = 0
+      while (i < c && activity == null) {
+        activity = getActivity(view.getChildAt(i).context)
```

Comment:
Why can't we just use the ViewGroup's context? This extra iteration is a bit confusing.

Reply:
I did try to use the viewgroup's content but that was leading to `NullPointerException` error and therefore keeping it as it is for now.